### ISSUE TITLE
global mesh inversion problem when visualizing

### ISSUE THREAD

**Issue by zearo-github** [g-fiche/VQ-HPS#8]
I'm encountering an interesting visualization issue while working with the HMR implementation in vq_hps/data/dataset_hmr.py and would appreciate your insights.

**Observed Phenomenon:**
When visualizing the global mesh (item["mesh"] = torch.from_numpy(global_mesh[0]).float() in vq_hps/data/dataset_hmr.py)

The rendered mesh appears inverted on the vertical axis. 

But the lcoal mesh is normal, just facing differently than the person in the picture.

The first line is the local mesh, the second line is the global mesh, and the third line is the image.

![Image](https://github.com/user-attachments/assets/7f008d13-8aeb-4c92-9f5c-ae75a3a85070)

**Technical Context:**
From my understanding, the primary distinction between local and global meshes lies in whether the get_body_model_sequence function utilizes root_orient for root orientation adjustment.

**Specific Queries:**
1.Could you help identify potential causes for this vertical inversion in global mesh visualization?

2.Does the above phenomenon not affect the prediction of the rotation matrix? 

3.Are there recommended best practices for ensuring consistent coordinate system alignment between mesh representations and input images?

Thank you for maintaining this valuable repository. I'd be grateful for any clarification you can provide regarding these mesh orientation behaviors

The visualization function used is based on: plot_meshes in mesh_vq_vae/utils/mesh_render:
`def plot_meshes_with_imgs(
        meshes,
        faces,
        device,
        raw_imgs,  # 新增输入参数
        show: bool = True,
        save: str = None,
        rot: bool = False,
        colors=None,
):
    """
    Plot a set of meshes and their corresponding images.
    """
    # 渲染网格图像
    images = renderer(meshes, faces, device, rot=rot, colors=colors)

    # 设置图形大小 (ncols * 224, nrows * 224)
    ncols = len(meshes)  # Number of meshes will determine the columns
    fig = plt.figure(figsize=(ncols * 224 / 100, 2 * 224 / 100))  # 转换为英寸（1英寸=100像素）

    # 创建网格布局
    gs = GridSpec(nrows=2, ncols=ncols)
    i = 0

    # 绘制mesh
    for col in range(ncols):
        ax = fig.add_subplot(gs[0, col])
        if images[i].shape[0] == 1:
            ax.imshow(images[i][0, :, :].cpu().detach().numpy())
        else:
            ax.imshow(images[i].cpu().detach().numpy())
        ax.axis("off")
        i += 1

    j = 0
    # 在底部显示图像
    for col in range(ncols):
        ax = fig.add_subplot(gs[1, col])
        # img = (raw_imgs[j].detach().numpy() * 255).astype(np.uint8).transpose(1, 2, 0)  # 转置为 (224, 224, 3)
        img = (raw_imgs[j].detach().numpy()).astype(np.uint8)  # 转置为 (224, 224, 3)
        ax.imshow(img)
        ax.axis('off')
        j += 1

    if show:
        plt.show()
    if save is not None:
        plt.savefig(save, bbox_inches='tight')
        plt.close()`

**Comment by g-fiche** [comment]
Hello,

The visualization will be satisfactory if you set the parameter ```rot=True``` when visualizing the global mesh. I think this is because the axis convention used for visualization differs from the one used in 3DPW annotations. 
We could have defined a different rotation for the canonical coordinates system so that we always rotate the mesh and do not have this problem, but we found it easier to set it to 0.

If you have a look at the code of other models, you will see that they also apply a rotation around the vertical axis; for instance, in [BEDLAM](https://github.com/pixelite1201/BEDLAM/blob/master/train/utils/mesh_viewer.py#L97) or in [HMR2.0](https://github.com/shubham-goel/4D-Humans/blob/main/hmr2/utils/renderer.py#L200C9-L201C40).

Please let me know if that solves your problem!

**Comment by zearo-github** [comment]
Thank you very much for your answer!

When I set the rot option to True, the visualizations did return to normal.

![Image](https://github.com/user-attachments/assets/1cd5922c-a941-416c-b8c0-07e5ef27073d)

**Comment by g-fiche** [comment]
Ok, great; thank you for your feedback!
